### PR TITLE
API Pod - Kubernetes Pod deployment of API

### PR DESCRIPTION
**Pull request by pataruco**
# API Pod - Kubernetes Pod deployment of API

[Airtable ticket](https://airtable.com/tbl7CISHKkr8Kdeh2/viwlDIU6cp8cbQYhj/recGQ019oeKIe7vmB?blocks=hide)

## Acceptance Criteria

- [x] A healthy graphql server running in Kubernetes pod

### Prerequisites

- Kubernetes [`kubectl`](https://kubernetes.io/docs/tasks/tools/install-kubectl/) to manage the Kubernetes cluster.
- Kubernetes [`stern`](https://github.com/wercker/stern)

### Deploy API pod in Kuberbetes cluster ⎈

1. Navigate to this directory, `/medicines/api`
2. Create [Kubernetes Service](https://kubernetes.io/docs/concepts/services-networking/service/)

   ```sh
   kubectl apply -f ./infrastructure/non-prod/deployment.yml
   ```

3. Deploy [Kubernetes pod](https://kubernetes.io/docs/concepts/workloads/pods/pod/) by applying a deployment

   ```sh
   kubectl apply -f ./infrastructure/non-prod/deployment.yml
   ```

4. Check if the pods are running

   ```sh
   stern api
   ```

5. You should reeceive a health server log like this

   ```sh
   api-558646c969-9mdxp api [2020-01-20T15:02:57Z INFO  actix_web::middleware::logger] 10.244.1.1:51524 "GET /healthz HTTP/1.1" 200 2 "-" "kube-probe/1.14" 0.000059
   ```

### CODE REVIEW

File: medicines/api/infrastructure/non-prod/deployment.yml [hunk 1]
```diff
@@ -15,14 +15,21 @@ spec:
...
+        env: non-prod
     spec:
       containers:
-        - image: api
+        - image: mhraproductsnonprod.azurecr.io/products/api
           name: api
+          readinessProbe:
+            httpGet:
+              path: /healthz
+              port: 8000
```
Comment (StuartHarris) [inline medicines/api/infrastructure/non-prod/deployment.yml]: if this port is the same as the main port, you can omit it

File: medicines/api/README.md [hunk 2]
```diff
@@ -26,3 +32,30 @@ To see the graphql explorer, go to http://127.0.0.1:8080/graphiql.
 3. Run `docker run -p 8080:8000 api`
 4. Open the browser and go to `http://0.0.0.0:8000/healthz`
```
Comment (StuartHarris) [inline medicines/api/README.md]: `http://localhost:8080/healthz`

File: medicines/api/README.md [hunk 3]
```diff
@@ -26,3 +32,30 @@ To see the graphql explorer, go to http://127.0.0.1:8080/graphiql.
...
 5. Yo should see an **Ok** and a server log in your terminal
+
+## Deploy API pod in Kuberbetes cluster ⎈
+
+1. Navigate to this directory, `/medicines/api`
+2. Create [Kubernetes Service](https://kubernetes.io/docs/concepts/services-networking/service/)
+
+   ```sh
+   kubectl apply -f ./infrastructure/non-prod/deployment.yml
```
Comment (StuartHarris) [inline medicines/api/README.md]: `kubectl apply -f ./infrastructure/non-prod/service.yml`

File: medicines/api/README.md [hunk 4]
```diff
@@ -26,3 +32,30 @@ To see the graphql explorer, go to http://127.0.0.1:8080/graphiql.
...
+   ```
+
+4. Check if the pods are running
+
+   ```sh
+   stern api
+   ```
+
+5. You should reeceive a server log like this
```
Comment (StuartHarris) [inline medicines/api/README.md]: typo

File: medicines/api/infrastructure/non-prod/deployment.yml (hunk 1 above)
Comment (pataruco) [reply medicines/api/infrastructure/non-prod/deployment.yml]: This field is required

File: medicines/api/README.md (hunk 2 above)
Comment (pataruco) [reply medicines/api/README.md]: Thanks, I've changed

File: medicines/api/README.md (hunk 3 above)
Comment (pataruco) [reply medicines/api/README.md]: Thanks, I've changed

File: medicines/api/README.md (hunk 4 above)
Comment (pataruco) [reply medicines/api/README.md]: Thanks, I've amended

**Pull request merged**